internal/util/diff/diff.go

```diff
@@ -294,14 +293,24 @@ func (pg defaultPkgGetter) GetPkg(repo, path, ref string) (string, error) {
 	if err != nil {
 		return dir, err
 	}
-	cmdGet := &fetch.Command{
-		RepoSpec: &git.RepoSpec{
-			OrgRepo: repo,
-			Ref:     ref,
-			Path:    path,
+
+	name := filepath.Base(dir)
+	kf := kptfileutil.DefaultKptfile(name)
+	kf.Upstream = &kptfilev1alpha2.Upstream{
+		Type: kptfilev1alpha2.GitOrigin,
+		Git: &kptfilev1alpha2.Git{
+			Repo:      repo,
+			Directory: "/",
+			Ref:       ref,
 		},
-		Destination: dir,
-		Clean:       true,
+	}
+	err = kptfileutil.WriteFile(dir, kf)
```

Comment:
Probably want a WriteKptFile() method to Package as counterpart to:

https://github.com/GoogleContainerTools/kpt/blob/f652b15b033a763967152e397d05320fe53166f4/internal/pkg/pkg.go#L91

Reply:
Yeah, we already have code for handling the Kptfile in `pkg/kptfile`, including `ReadFile` and `WriteFile` functions. But we seem to also have added this in `internal/pkg`.

Reply:
General note: having `fooutil` and `barutil` is usually a code smell. Aside from `util` being a meaningless term (Go naming practices advise against it), it usually signals that there we don't have the right abstractions in place.

Reply:
Agree there are too many `util` packages in kpt now. We should create more focused packages and remove the duplication.